testsuites/syncgateway/functional/tests/test_xattrs.py

```diff
@@ -1212,6 +1213,257 @@ def update_props():
+
+    sg_client.create_user(url=sg_admin_url, db=sg_db, name='autosguser', password='pass', channels=['sg-shared'])
+    autosguser_session = sg_client.create_session(url=sg_admin_url, db=sg_db, name='autosguser', password='pass')
+
+    log_info('Num docs per client: {}'.format(number_docs_per_client))
+    log_info('Num updates per doc per client: {}'.format(number_updates_per_doc_per_client))
+
+    log_info('sg_conf: {}'.format(sg_conf))
+    log_info('sg_admin_url: {}'.format(sg_admin_url))
+    log_info('sg_url: {}'.format(sg_url))
+
+    sg_tracking_prop = 'sg_one_updates'
+    sdk_tracking_prop = 'sdk_one_updates'
+
+    # Start listening to changes feed
+    changestrack = ChangesTracker(sg_url, sg_db, auth=autosdkuser_session)
+    changestrack_sg = ChangesTracker(sg_url, sg_db, auth=autosguser_session)
+    cbs_ip = host_for_url(cbs_url)
+
+    # Connect to server via SDK
+    sdk_client = Bucket('couchbase://{}/{}'.format(cbs_ip, bucket_name), password='password', timeout=SDK_TIMEOUT)
+
+    # Inject custom properties into doc template
+    def update_props():
+        return {
+            'updates': 0,
+            sg_tracking_prop: 0,
+            sdk_tracking_prop: 0
+        }
+
+    # Create / add docs via sdk
+    sdk_doc_bodies = document.create_docs(
+        'doc_sdk_ids',
+        number_docs_per_client,
+        channels=['shared'],
+        prop_generator=update_props
+    )
+
+    with ThreadPoolExecutor(max_workers=5) as tpe:
+
+        # Add docs via SDK
+        log_info('Started adding {} docs via SDK ...'.format(number_docs_per_client))
+        sdk_docs = {doc['_id']: doc for doc in sdk_doc_bodies}
+        doc_set_ids1 = [sdk_doc['_id'] for sdk_doc in sdk_doc_bodies]
+        sdk_docs_resp = sdk_client.upsert_multi(sdk_docs)
+        assert len(sdk_docs_resp) == number_docs_per_client
+        assert len(doc_set_ids1) == number_docs_per_client
+        log_info("Docs creation via SDK done")
+        all_docs_via_sg_formatted = [
+            {"id": doc, "rev": "1-"} for doc in doc_set_ids1]
+
+        ct_task = tpe.submit(changestrack.start(timeout=10))
+        log_info("ct_task value {}".format(ct_task))
+        wait_for_changes = tpe.submit(
+            changestrack.wait_until, all_docs_via_sg_formatted, rev_prefix_gen=True)
+
+        if wait_for_changes.result():
+            log_info("Found all docs ...")
+        else:
+            raise Exception(
+                "Could not find all changes in feed for adding docs via SDK before timeout!!")
+
+    with ThreadPoolExecutor(max_workers=5) as tpe:
+        log_info("Updating docs via SDK...")
+
+        # Update docs via SDK
+        sdk_docs = sdk_client.get_multi(doc_set_ids1)
+        assert len(sdk_docs.keys()) == number_docs_per_client
+        for doc_id, val in sdk_docs.items():
+            log_info("Updating: '{}' via SDK".format(doc_id))
```

Comment:
Actually this is used for my debugging. I removed this as there are enough logging already and unnecessarity console logs this, so removed it